account_members.go

```diff
@@ -81,18 +83,73 @@ func (api *API) AccountMembers(ctx context.Context, accountID string, pageOpts P
 //
 // API reference: https://api.cloudflare.com/#account-members-add-member
 func (api *API) CreateAccountMemberWithStatus(ctx context.Context, accountID string, emailAddress string, roles []string, status string) (AccountMember, error) {
+	invite := AccountMemberInvitation{
+		Email:    emailAddress,
+		Roles:    roles,
+		Policies: nil,
+		Status:   status,
+	}
+	return api.CreateAccountMemberInternal(ctx, accountID, invite)
+}
+
+// CreateAccountMember invites a new member to join an account with roles.
+// The member will be placed into "pending" status and receive an email confirmation.
+// NOTE: If you are currently enrolled in Domain Scoped Roles, your roles will be converted to policies
+// upon member invitation. We recommend upgrading to CreateAccountMemberWithPolicies to use policies.
+//
+// API reference: https://api.cloudflare.com/#account-members-add-member
+func (api *API) CreateAccountMember(ctx context.Context, accountID string, emailAddress string, roles []string) (AccountMember, error) {
```

Comment:
if we're overhauling the invitations, let's update `CreateAccountMember` method signature to match the [experimental client](https://github.com/cloudflare/cloudflare-go/blob/master/docs/experimental.md). this will also allow you to cut down on some of the boilerplate code we're repeating and putting into `CreateAccountMemberInternal`.

(incomplete but hopefully enough to get you started example)

```
type CreateAccountMemberParams struct {
	Email    string   `json:"email"`
	Roles    []string `json:"roles,omitempty"`
	Policies []Policy `json:"policies,omitempty"`
	Status   string   `json:"status,omitempty"`
}

func (api *API) CreateAccountMember(ctx context.Context, rc *ResourceContainer, params CreateAccountMemberParams) (AccountMember, error) {
	invite := AccountMemberInvitation{
		Email: params.Email,
	}

	if len(params.Roles) > 0 {
		invite.Roles = params.Roles
	}

	// ... etc
}
```

Reply:
Good call, thanks! Can you confirm that this is the right way to go about things?
- Re-write `CreateAccountMember` to match the experimental client
- Remove the `CreateAccountMemberInternal` method
- Re-factor utility methods (`CreateAccountMemberWithStatus`, `CreateAccountMemberWithRoles`, etc) to just call `CreateAccountMember` - the alternative I supposed is just removing the utility methods?

Here's my current `CreateAccountMember` impl, let me know if it looks okay and the above is correct and I'll push :) 
```go
func (api *API) CreateAccountMember(ctx context.Context, rc *ResourceContainer, params CreateAccountMemberParams) (AccountMember, error) {
	if rc.Level != AccountRouteLevel {
		return AccountMember{}, fmt.Errorf(errInvalidResourceContainerAccess, rc.Level)
	}

	invite := AccountMemberInvitation{
		Email:  params.EmailAddress,
		Status: params.Status,
	}

	roles := []AccountRole{}
	for i := 0; i < len(params.Roles); i++ {
		roles = append(roles, AccountRole{ID: invite.Roles[i]})
	}
	err := validateRolesAndPolicies(roles, params.Policies)
	if err != nil {
		return AccountMember{}, err
	}

	if params.Roles != nil {
		invite.Roles = params.Roles
	} else if params.Policies != nil {
		invite.Policies = params.Policies
	}

	uri := fmt.Sprintf("/accounts/%s/members", rc.Identifier)
	res, err := api.makeRequestContext(ctx, http.MethodPost, uri, invite)
	if err != nil {
		return AccountMember{}, err
	}

	var accountMemberListResponse AccountMemberDetailResponse
	err = json.Unmarshal(res, &accountMemberListResponse)
	if err != nil {
		return AccountMember{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
	}

	return accountMemberListResponse.Result, nil
}
```

Reply:
looks spot on! nice one.